test/models/lookup_key_test.rb

```diff
@@ -420,4 +420,24 @@ def setup
       assert_valid lookup_key
     end
   end
+
+  test "can update lookup key with default_value without audit error" do
+    location_list = [taxonomies(:location1), taxonomies(:location2)]
+    puppetclass1 = FactoryBot.create(:puppetclass, :with_auditing)
+    environments = FactoryBot.create_list(:environment, 3)
+    environments.each do |e|
+      e.locations = location_list
+      e.puppetclasses = [puppetclass1]
+      e.save!
+    end
+    puppetclass1.reload
+    puppetclass1.update_attribute(:name, 'xyz')
+    audit_record = puppetclass1.audits.find_by(:action => 'update')
+    if audit_record
```

Comment:
This question remains open?

Reply:
The only reason why I have added this check is because in the past we have seen many random jobs failures related to audit_records whereas those used to run locally without any failure.

Reply:
I am OK to remove it but I would like to know your suggestion.

Reply:
Lets remove it and see if this causes failures, I'm worried we might miss failures due to the audit not being loaded.

Reply:
Removed! :)